compliance/collection/main.go

```diff
@@ -152,15 +164,22 @@ func manageReceiveStream(ctx context.Context, cli sensor.ComplianceServiceClient
 
 func manageNodeScanLoop(ctx context.Context, rescanInterval time.Duration, scanner nodeinventorizer.NodeInventorizer) <-chan *sensor.MsgFromCompliance {
 	sensorC := make(chan *sensor.MsgFromCompliance)
-	nodeName := getNode()
+	inventoryOpts := &nodeinventorizer.InventoryScanOpts{
+		NodeName:            getNode(),
+		Scanner:             scanner,
+		InventoryCachePath:  "/cache/inventory-cache",
+		BackoffFilePath:     "/cache/inventory-backoff",
+		BackoffWaitCallback: nodeInventoryWaitCallback,
+	}
+
 	go func() {
 		defer close(sensorC)
 		t := time.NewTicker(rescanInterval)
 
 		// first scan should happen on start
-		msg, err := scanNode(nodeName, scanner)
+		msg, err := nodeinventorizer.TriggerNodeInventory(inventoryOpts)
```

Comment:
I was considering building a `CachingScanner`, but decided against it to keep the Fake and the real scanner as close as possible.
But thinking about it again, it might be the right call as opposed to layering something on the Inventorizer instead.

Reply:
I have reworked this to an implementation of NodeInventorizer in c57265d7034e5a4cf409a3e458b9d9d5b29e3972, please TAL.